pkg/scheduling/requirement.go

```diff
@@ -77,11 +78,21 @@ func NewRequirementWithFlexibility(key string, operator corev1.NodeSelectorOpera
 	}
 	if operator == corev1.NodeSelectorOpGt {
 		value, _ := strconv.Atoi(values[0]) // prevalidated
-		r.greaterThan = &value
+		value++                             // canonicalize GT N to GTE N+1
```

Comment:
We should probably use CEL to bound the LT and GT values such that this doesn't result in an overflow (or an underflow for other operations). FWIW though I can't really imagine a use case where this would be an issue considering the requirement would be nonsense (e.g. GT MAX_INT or LT MIN_INT).